Android/GreenCircle/app/src/main/java/com/greencircle/framework/views/fragments/company/upload_documents/UploadDocumentDialogFragment.kt

```diff
@@ -0,0 +1,120 @@
+        arguments = requireArguments()
+        authToken = arguments.getString("authToken").toString()
+        companyId = arguments.getString("companyId")
+        Log.d("CUstomSucc2", companyId.toString())
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val inflater = requireActivity().layoutInflater
+        view = inflater.inflate(R.layout.modal_upload_document, null)
+        viewModel = UploadFilesViewModel(requireContext())
+
+        val cardView = view.findViewById<CardView>(R.id.uploadFileCard)
+        cardView.setOnClickListener {
+            launchPicker()
+        }
+        val cancelFileButton = view.findViewById<Button>(R.id.cancelFileButton)
+        cancelFileButton.setOnClickListener {
+            dialog?.dismiss()
+        }
+
+        val builder = AlertDialog.Builder(requireActivity())
+        builder.setView(view)
+            .setTitle(title)
+            .setMessage("Sube tu archivo aqui")
+        return builder.create()
+    }
+
+    private fun launchPicker() {
+        val intent = Intent(Intent.ACTION_GET_CONTENT)
+        intent.type = "application/pdf"
+        startActivityForResult(intent, 1)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (resultCode == RESULT_OK && data != null) {
+            val fileUri: Uri? = data.data
+            fileUri?.let {
+                val filePath = getFilePath(it)
+                fileToUpload = File(filePath)
+                val fileName = fileToUpload.name
+                changeDialogAfterSelection(fileName)
+
+                isFileSelected = true
+                val uploadFileButton = view.findViewById<Button>(R.id.uploadFileButton)
+                if (isFileSelected) {
+                    uploadFileButton.isEnabled = true
+                    uploadFileButton.setOnClickListener {
+                        uploadFile(fileToUpload)
+                        dialog?.dismiss()
+                    }
+                }
+            }
+        }
+    }
+
+    private fun getFilePath(uri: Uri): String? {
+        val projection = arrayOf(OpenableColumns.DISPLAY_NAME)
+        val cursor = requireContext().contentResolver.query(uri, projection, null, null, null)
+        cursor?.use {
+            val nameIndex = it.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+            if (it.moveToFirst()) {
+                return File(requireContext().cacheDir, it.getString(nameIndex)).absolutePath
+            }
+        }
+        return null
+    }
+
+    private fun changeDialogAfterSelection(fileName: String) {
+        dialog?.findViewById<TextView>(R.id.selectedFileName)?.text = fileName
+        dialog?.findViewById<ImageView>(R.id.uploadFileImage)?.visibility = View.GONE
+        dialog?.findViewById<ImageView>(R.id.checkmarkImageView)?.visibility = View.VISIBLE
+    }
+
+    private fun uploadFile(file: File) {
+        // TODO: Change to real companyId
```

Comment:
Hay algun motivo porque esta funcion tiene el comment de TODO: Change to real company ID?